packages/runtime/container-runtime/src/summaryFormat.ts

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { SummaryType } from "@fluidframework/protocol-definitions";
+import { channelsTreeName, ISummaryTreeWithStats } from "@fluidframework/runtime-definitions";
+import { IFluidDataStoreAttributes } from "./dataStoreContext";
+
+export type ContainerRuntimeSummaryFormatVersion =
+    /**
+     * Version 0: format version is missing from summary.
+     * This indicates it is an older version.
+     */
+    | undefined
+    /**
+     * Introduces .metadata blob and .channels trees for isolation of
+     * data store trees from container-level objects.
+     */
+    | 1;
+
+export type DataStoreSummaryFormatVersion =
+    /**
+     * Version 0: format version is missing from summary.
+     * This indicates it is an older version.
+     */
+    | undefined
+    /**
+     * Version 1: from this version the pkg within the data store
+     * attributes blob is a JSON array rather than a string.
+     */
+    | "0.1"
+    /**
+     * Introduces .channels trees for isolation of
+     * channel trees from data store objects.
+     */
+    | 2;
+
+export function summaryFormatVersionToNumber(
+    version: ContainerRuntimeSummaryFormatVersion | DataStoreSummaryFormatVersion,
+): number {
+    if (version === undefined) {
+        return 0;
+    }
+    if (version === "0.1") {
+        return 1;
+    }
+    return version;
+}
+
+export const metadataBlobName = ".metadata";
+export const chunksBlobName = ".chunks";
+export const blobsTreeName = ".blobs";
+
+export interface IContainerRuntimeMetadata {
+    readonly summaryFormatVersion: ContainerRuntimeSummaryFormatVersion;
```

Comment:
I just realized that this is called `summaryFormatVersion` whereas the data store one is called `snapshotFormatVersion`. I know its going to be a back-compat nightmare, but we should try and move towards making this naming consistent.

Reply:
Thoughts? Make this one `snapshotFormatVersion` or work on migrating the other one to `summaryFormatVersion`?

Reply:
Since we are moving away from snapshot terminology, I would rather have `summaryFormatVersion` everywhere.

Reply:
Okay but we will need to support reading `snapshotFormatVersion` forever :)

Reply:
Yeah I know. Hence I called this a back-compat nightmare :-). I do not have very strong feeling towards it though. Maybe create an issue for now and we can evaluate. Doesn't have to be in this change.

Reply:
I made this change, feel free to review.